farm/modeling/adaptive_model.py

```diff
@@ -690,7 +691,7 @@ class ONNXAdaptiveModel(BaseAdaptiveModel):
     For inference, this class is compatible with the FARM Inferencer.
     """
     def __init__(self, onnx_session, prediction_heads, language, device="cpu"):
-        if str(onnxruntime.get_device()).lower() != str(device).lower():
+        if device == "cuda" and onnxruntime.get_device() != "gpu":
```

Comment:
should device here really be a string or rather a `torch.device` as returned by `intialize_device_settings()`?

Reply:
Yes, it is intended to be `torch.device`. Also, the assertion is now simplified and the `device` argument is compliant with that of `AdaptiveModel`.